File: CodeWars/8kyu/JavaScript/001-get-the-mean-of-an-array.js
// https://www.codewars.com/kata/563e320cee5dddcf77000158
// It's the academic year's end, fateful moment of your school report. The averages must be calculated. All the students come to you and entreat you to calculate their average for them. Easy ! You just need to write a script.
// Return the average of the given array rounded down to its nearest integer.
// The array will never be empty.

function getAverage(marks) {

    let sum = 0;

    //iterate over the array and sum it
    for (let i = 0; i < marks.length; i++) {
        sum += marks[i];
    }

    let result = Math.floor(sum / marks.length);

    return result
}

function getAverage(marks) {
    //use reduce
    let sum = marks.reduce((sum, current) => {
        return sum += current
    })

    let result = Math.floor(sum / marks.length);

    return result
}

console.log(getAverage([2, 2, 2, 2]), 2);
console.log(getAverage([1, 2, 3, 4, 5,]), 3);
console.log(getAverage([1, 1, 1, 1, 1, 1, 1, 2]), 1);

File: CodeWars/8kyu/Python/001-get-the-mean-of-an-array.py
# https://www.codewars.com/kata/563e320cee5dddcf77000158/
# Return the average of the given array rounded down to its nearest integer.
# The array will never be empty.

import math


def get_average(marks):
    sum = 0
    for i in range(len(marks)):
        sum += marks[i]

    return math.floor(sum/len(marks))


def get_average(marks):
    sum = 0
    for num in marks:
        sum += num

    return math.floor(sum/len(marks))


def get_average(marks):
    sum = 0
    for num in marks:
        sum += num

    return math.floor(sum/len(marks))


def get_average(marks):
    return int(sum(marks) / len(marks))

def get_average(marks):
    return sum(marks) // len(marks)


import numpy
def get_average(marks):
    return int(numpy.mean(marks))


get_average =lambda x: sum(x)//len(x)

print(get_average([2, 2, 2, 2]), 2)
print(get_average([1, 5, 87, 45, 8, 8]), 25)
print(get_average([2, 5, 13, 20, 16, 16, 10]), 11)
print(get_average([1, 2, 15, 15, 17, 11, 12,
                   17, 17, 14, 13, 15, 6, 11, 8, 7]), 11)
